python/src/execution.rs

```diff
@@ -13,15 +13,15 @@ use crate::error::{ErrorContext, IntoError, Result};
 #[pyclass]
 #[derive(Clone)]
 pub(crate) struct Execution {
-    execution: Arc<Mutex<Option<RustExecution>>>,
+    execution: Arc<Option<RustExecution>>,
```

Comment:
Given the difficulties, it seems like we may be better off having submitting an execution return a pair of `Execution, StopChannel`. Then we can leave the mutex in the execution code, but interact with the stop channel without locking it. Thoughts?

Reply:
Yeah, I guess so 😞 . I'll try that route